src/main/kotlin/uk/gov/justice/digital/hmpps/sentenceplan/entity/PlanEntity.kt

```diff
@@ -76,6 +77,14 @@ enum class PublishState {
 interface PlanRepository : JpaRepository<PlanEntity, Long> {
   fun findByUuid(planUuid: UUID): PlanEntity
 
+  @Query(
+    """
+    select p from PlanEntity p
+    where p.uuid = :planUuid
+  """,
+  )
+  fun findByPlanId(planUuid: UUID): PlanEntity?
```

Comment:
is the only difference with the method above the potential null return?

Reply:
If so can we just call it something more descriptive like `findPlanByUuid`, since it is not finding the Plan by the ID field

Reply:
Yeah - find it easier to just return null and throw the execption in an extension function if its null. That way we dont need to worry about it after that. Renamed